File: data-pipeline/user-cache-updater/src/main/java/org/ekstep/ep/samza/domain/Event.java
package org.ekstep.ep.samza.domain;

import org.ekstep.ep.samza.reader.NullableValue;
import org.ekstep.ep.samza.reader.Telemetry;

import java.util.ArrayList;
import java.util.Map;

public class Event {
    private final Telemetry telemetry;

    public Event(Map<String, Object> map) {
        this.telemetry = new Telemetry(map);
    }

    public String objectUserId() {
        NullableValue<String> objectType = telemetry.read("object.type");
        NullableValue<String> objectUserId = null;
        if (null != objectType.value() && objectType.value().equalsIgnoreCase("User")) {
            objectUserId = telemetry.read("object.id");
        }
        return null!=objectUserId ? objectUserId.value(): null;
    }

    public String getUserSignInType() {
        NullableValue<ArrayList<Map<String, String>>> cdata = telemetry.read("context.cdata");
        ArrayList<Map<String, String>> cdataList = cdata.value();
        if (null != cdataList && !cdataList.isEmpty()) {
            for (Map<String, String> cdataMap : cdataList) {
                if (cdataMap.containsKey("type") && cdataMap.get("type").equalsIgnoreCase("SignupType"))
                    return cdataMap.get("id").toString();
            }
        }
        return null;
    }

    public String getUserLoginType() {
        NullableValue<ArrayList<Map<String, String>>> cdata = telemetry.read("context.cdata");
        ArrayList<Map<String, String>> cdataList = cdata.value();
        if (null != cdataList && !cdataList.isEmpty()) {
            for (Map<String, String> cdataMap : cdataList) {
                if (cdataMap.containsKey("type") && cdataMap.get("type").equalsIgnoreCase("UserRole"))
                    return cdataMap.get("id").toString();
            }
        }
        return null;
    }

    public ArrayList<String> getUserMetdataUpdatedList() {
        NullableValue<ArrayList<String>> edata_props = telemetry.read("edata.props");
        if(null != edata_props) {
            ArrayList<String> edata_props_list = edata_props.value();
            if(null != edata_props_list && !edata_props_list.isEmpty()) {
                return edata_props_list;
            }
        }
        return null;
    }

    public String getUserStateValue() {
        NullableValue<String> edata_state = telemetry.read("edata.state");
        if( null != edata_state.value() && edata_state.value().equalsIgnoreCase("Create") || edata_state.value().equalsIgnoreCase("Created")) {
            return "Create";
        }
        else if( null != edata_state.value() && edata_state.value().equalsIgnoreCase("Update") || edata_state.value().equalsIgnoreCase("Updated")) {
            return "Update";
        }
        return null;
    }

}

File: data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/domain/Event.scala
package org.sunbird.dp.usercache.domain

import java.util

import org.sunbird.dp.core.domain.Events
import org.sunbird.dp.usercache.util.UserReadResult

class Event(eventMap: util.Map[String, Any]) extends Events(eventMap) {

  override def kafkaKey(): String = {
    did()
  }

  def getId: String = {
    Option(objectType()).map({ t => if (t.equalsIgnoreCase("User")) objectID() else null
    }).getOrElse(null)
  }

  def getState: String = {
    telemetry.read[String]("edata.state").getOrElse(null)
  }

  def getContextDataId(cDataType: String): String = {
    val cdata = telemetry.read[util.ArrayList[util.Map[String, AnyRef]]]("context.cdata").getOrElse(null)
    var signInType: String = null
    Option(cdata).map(data => {
      data.forEach(cdataMap => {
        if (cdataMap.get("type").asInstanceOf[String].equalsIgnoreCase(cDataType)) signInType = cdataMap.get("id").toString else signInType
      })
    }).getOrElse(signInType)
    signInType
  }

  def userMetaData(): util.ArrayList[String] = {
    telemetry.read[util.ArrayList[String]]("edata.props").getOrElse(new util.ArrayList[String]())
  }

  def isValid(userReadRes: UserReadResult) = {
    if (userReadRes.responseCode.toUpperCase.equalsIgnoreCase("OK") && !userReadRes.result.isEmpty && userReadRes.result.containsKey("response")) true else false
  }

}
